### PR TITLE
refactor: remove allowlist in GetSummary of scan controller

### LINKED ISSUE #13919: Remove the CVE allowlist option in the scan summary processing
We need to remove the CVE allow list option in the scan summary processing to make ensure the summary indicates the real summary for the artifact.

For vulnerability prevent, the middleware can compute the severity from the vulnerabilities of the report and CVE allowlist of the project.

### PR DESCRIPTION
**Pull request by heww**
Closes #13919

Signed-off-by: He Weiwei <hweiwei@vmware.com>

### CODE REVIEW

File: src/controller/scan/base_controller.go [hunk 1]
```diff
@@ -699,6 +701,82 @@ func (bc *basicController) DeleteReports(ctx context.Context, digests ...string)
 	return nil
 }
 
+func (bc *basicController) GetVulnerable(ctx context.Context, artifact *ar.Artifact, allowlist allowlist.CVESet) (*Vulnerable, error) {
+	var (
```
Comment (wy65701436) [inline src/controller/scan/base_controller.go:705]: add artifact != nil check?

File: src/controller/scan/base_controller.go [hunk 2]
```diff
@@ -699,6 +701,82 @@ func (bc *basicController) DeleteReports(ctx context.Context, digests ...string)
...
+
+				continue
+			}
+
+			if severity == "" || v.Severity.Code() > severity.Code() {
+				severity = v.Severity
+			}
+		}
+
+		if severity != "" {
```
Comment (wy65701436) [inline src/controller/scan/base_controller.go:772]: what about else?

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/controller/scan/base_controller.go [hunk 3]
```diff
@@ -699,6 +701,82 @@ func (bc *basicController) DeleteReports(ctx context.Context, digests ...string)
...
 
+func (bc *basicController) GetVulnerable(ctx context.Context, artifact *ar.Artifact, allowlist allowlist.CVESet) (*Vulnerable, error) {
+	var (
+		mimeType string
+		reports  []*scan.Report
+	)
+	for _, m := range []string{v1.MimeTypeNativeReport, v1.MimeTypeGenericVulnerabilityReport} {
+		rps, err := bc.GetReport(ctx, artifact, []string{m})
+		if err != nil {
+			log.G(ctx).Errorf("get vulnerability report of the artifact %s@%s failed, error: %v", artifact.RepositoryName, artifact.Digest, err)
```
Comment (steven-zou) [inline src/controller/scan/base_controller.go:712]: Why do you have to print the log here as the error has been returned to the upper caller?  It seems the log can be printed in an upper hander.

File: src/pkg/scan/vuln/util.go [hunk 4]
```diff
@@ -79,12 +79,14 @@ func mergeSeverity(s1, s2 Severity) Severity {
...
 
-func mergeScanStatus(s1, s2 string) string {
+// MergeScanStatus ...
+func MergeScanStatus(s1, s2 string) string {
 	j1, j2 := job.Status(s1), job.Status(s2)
 
 	if j1 == job.RunningStatus || j2 == job.RunningStatus {
 		return job.RunningStatus.String()
 	} else if j1 == job.SuccessStatus || j2 == job.SuccessStatus {
+		// the scan status of the image index will be treated as a success when one of its children is success
```
Comment (steven-zou) [inline src/pkg/scan/vuln/util.go:89]: From the overall summary, it seems to show partial data to users.

Maybe in the future, we can render a report summary for each of the platforms indexed.

**Review by steven-zou — COMMENTED** [review]
comments

File: src/pkg/scan/vuln/util.go (hunk 4 above)
Comment (heww) [reply src/pkg/scan/vuln/util.go:89]: +1. I forgot why the status is successful when one of the reports is successful, so I added some comments here.

**Review by steven-zou — APPROVED** [review]
LGTM

**Pull request merged**